vector/src/main/java/im/vector/app/features/onboarding/OnboardingViewModel.kt

```diff
@@ -167,13 +170,47 @@ class OnboardingViewModel @AssistedInject constructor(
         }
     }
 
-    private fun handleMaybeUpdateHomeserver(action: OnboardingAction.MaybeUpdateHomeserverFromMatrixId) {
-        val isFullMatrixId = MatrixPatterns.isUserId(action.userId)
+    private fun handleUserNameEntered(action: OnboardingAction.UserNameEnteredAction) {
+        when (action) {
+            is OnboardingAction.UserNameEnteredAction.Login -> maybeUpdateHomeserver(action.userId)
+            is OnboardingAction.UserNameEnteredAction.Registration -> maybeUpdateHomeserver(action.userId, continuation = { userName ->
+                checkUserNameAvailability(userName)
+            })
+        }
+    }
+
+    private fun maybeUpdateHomeserver(userNameOrMatrixId: String, continuation: suspend (String) -> Unit = {}) {
```

Comment:
Not sure that the lint will be ok but I suggest to name the string parameter inside the lambda for clarity
```suggestion
    private fun maybeUpdateHomeserver(userNameOrMatrixId: String, continuation: suspend (userName: String) -> Unit = {}) {
```

Reply:
I agree with this remark, also having an optional lambda for `continuation` is a bit strange to me.
Adam may iterate later, we need to merge the PR and review the next one before tomorrow, and this is not a blocker.